cpu/kinetis_common/random_rnga.c

```diff
@@ -55,7 +55,7 @@ int random_read(char *buf, unsigned int num)
         }
     }
 
-    return count;
+    return (int)count;
```

Comment:
Yes, I assumed that. But at least theoretical this could lead to a problem, if you specify a crazy big value.